### PR TITLE
Add virtual machine manager

### PR DESCRIPTION
**Pull request by realtek2017**
Functionality extended to support Virtual Machine Manager API.

New methods include:

    List all Virtual Guest Names
    List Virtual Guest Statuses
    Power on a Guest
    Power off a Guest
    Shutdown a Guest

### CODE REVIEW

File: src/synology_dsm/api/virtual_machine_manager/guest.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+"""DSM Virtual Machine data."""
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoVirtualMachineManager:
```
Comment (mib1185) [inline src/synology_dsm/api/virtual_machine_manager/guest.py:5]: this class should be moved to `__init__.py`

File: src/synology_dsm/api/virtual_machine_manager/guest.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    def __init__(self, dsm):
+        """Constructor method."""
+        self._dsm = dsm
+        self._data = {}
+
+    def update(self):
+        """Updates Virtual Machine Guest data."""
+        raw_data = self._dsm.get(self.API_KEY, "list")
+        if raw_data:
+            self._data = raw_data["data"]
```
Comment (mib1185) [inline src/synology_dsm/api/virtual_machine_manager/guest.py:20]: you should create an own class for the guest object (_suggest `SynoVmmGuest`_) which holds all the data and exposes properties (_compare with surveillance_station_)

**Review by mib1185 — CHANGES_REQUESTED** [review]

**Comment by mib1185** [comment]
Hi @realtek2017 
I like this idea, so let's work on it to bring it on the road 😉 

Overall please compare with the implementation of surveillance_station - it implements the generic class `SynoSurveillanceStation` which more or less represents the app itself and the class `SynoCamera` which represents the single camera instance with its properties like the name, id, model,... 
Last but not least ... tests are also needed 😉

**Comment by realtek2017** [comment]
> Hi @realtek2017 I like this idea, so let's work on it to bring it on the road wink
> 
> Overall please compare with the implementation of surveillance_station - it implements the generic class `SynoSurveillanceStation` which more or less represents the app itself and the class `SynoCamera` which represents the single camera instance with its properties like the name, id, model,... Last but not least ... tests are also needed wink
> 
> Btw I'm missing the reboot method wink

Thanks @mib1185 for all your comments, I appreciate your feedback.

My reason for adding this was driven from being able to use this eventually on the homeassistant integration :)

I will make the changes as requested, had some issues with getting poetry working so I will try and fix that for the tests.

**Comment by mib1185** [comment]
> My reason for adding this was driven from being able to use this eventually on the homeassistant integration :)

was hoping, that this is the reason behind 🤞 

> I will make the changes as requested, had some issues with getting poetry working so I will try and fix that for the tests.

If you are using VSCode and have Docker installed on your system, you can use the devcontainer in this project 😉

**Comment by mib1185** [comment]
This will be implemented with #363

**Pull request closed without merging**